Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -543,19 +543,36 @@ bool HiresTexture::LoadTexture(Level& level, const std::vector<u8>& buffer)
 
 std::string HiresTexture::GetTextureDirectory(const std::string& game_id)
 {
-  const std::string texture_directory = File::GetUserPath(D_HIRESTEXTURES_IDX) + game_id;
+  const std::string cstex_altid = game_id.substr(0, 3);  // 3-character region-free version
+  const std::string cstex_rootdir = File::GetUserPath(D_HIRESTEXTURES_IDX);
+  const std::string cstex_bestdir = File::GetUserPath(D_HIRESTEXTURES_IDX) + game_id;
+  const std::string cstex_altdir = File::GetUserPath(D_HIRESTEXTURES_IDX) + cstex_altid;
 
   // If there's no directory with the region-specific ID, look for a 3-character region-free one
-  if (!File::Exists(texture_directory))
-    return File::GetUserPath(D_HIRESTEXTURES_IDX) + game_id.substr(0, 3);
+  if (!File::Exists(cstex_bestdir))
+  {
+    INFO_LOG(VIDEO, "Custom textures directory '%s' does not exist, attempting region-free version",
+             game_id.c_str());
 
-  return texture_directory;
-}
+    if (File::Exists(cstex_altdir))
+    {
+      return cstex_altdir;
+    }
+    else
+    {
+      OSD::AddMessage(StringFromFormat("Missing: No custom textures for %s or %s were found!",
```

Comment:
Please, no. I don't think people need to see an OSD message just because there are no custom textures. That's just useless noise.